lib/haml_lint/linter/id_names.rb

```diff
@@ -0,0 +1,20 @@
+module HamlLint
+  # Checks for `id` attributes in specific cases on tags.
+  class Linter::IdNames < Linter
+    include LinterRegistry
+
+    STYLE = {
+      'camel_case' => /\A[a-z][\da-zA-Z]+\z/,
+      'lisp_case' => /\A[\da-z-]+\z/,
+      'pascal_case' => /\A[A-Z][\da-zA-Z]+\z/,
+      'snake_case' => /\A[\da-z_]+\z/,
+    }.freeze
+
+    def visit_tag(node)
+      return unless (id = node.tag_id)
+
+      style = STYLE[config['style']] || STYLE['lisp_case']
+      record_lint(node, '`id` attribute must be in lisp-case') unless id =~ style
```

Comment:
You should replace "lisp-case" in the error message with something like `config['style'] || 'lisp-case'`